### DISCUSSION TITLE
Is there a way to perform case insensitive searches?

### DISCUSSION THREAD

**Discussion by hiro5id** [photoprism/photoprism#2002] [Q&A] [answered]
I noticed that when I search `filename:"*spawn*"` it doesn't return any results, but when I do `filename:"*Spawn*"` that works.

I'm guessing there is some place in the code that controls case sensitivity search.  perhaps at the database level?  Is there a way to ignore casing during searches?  There is no way I am going to remember how I capitalized the words to be able to find things again.  By default I tend to search for things all lower case.

(Great job by the way on this project.  Loving that it is open source self-hosted.)

**Comment by lastzero** [comment, marked as answer]
The index currently is binary for best performance. You can try to manually change it to unicode, which should lead to case insensitive comparisons. However, it can cause problems when a file name exists twice in upper and lower case.

**Reply by hiro5id** [reply]
Then perhaps an idea would be whenever indexing filenames/metadata, it could be indexed as all lowercased, so that it makes it more likely that when we search for something, there will be hits. 

Perhaps it could even be a preference flag, in case some people like the current case sensitivity.

**Reply by lastzero** [reply]
Most users have a case sensitive file system, so the index must be case sensitive too in general - except in specific environments such as macOS with a case insensitive formated filesystem.

**Reply by hiro5id** [reply]
Yea I understand... When it comes to file names and paths the PP references physically on the hard disk of course should be case sensitive no doubt.   But the index, the metadata that people search on should be case insensitive.   I'm pretty sure i'd be happy to see both results if I search for `picture` it would return all `Picture.jpg` and `picture.jpg` and `pictUre.jpg`.   That would actually be preferable to most people I would think.  Because it means that it will be more likely I will find what i'm looking for.
